src/figuresClasses.ts

```diff
@@ -1,11 +1,71 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
```

Comment:
[CODE STYLE] - The error message could be more descriptive by specifying what values were received and stating the exact requirement. For example: 'Triangle side lengths must be greater than 0. Received: a=${a}, b=${b}, c=${c}'